modules/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobContainerRetriesTests.java

```diff
@@ -173,6 +208,29 @@ StorageOptions createStorageOptions(
         return new GoogleCloudStorageBlobContainer(randomBoolean() ? BlobPath.EMPTY : BlobPath.EMPTY.add("foo"), blobStore);
     }
 
+    public void testShouldRetryOnNetworkOutage() {
+        final int maxRetries = randomIntBetween(3, 5);
+        // We can't connect to the server here
+        if (randomBoolean()) {
+            logger.info("Failing due to connection refused");
+            endpointUrlOverride = "http://127.0.0.1:"
+                + randomValueOtherThan(httpServer.getAddress().getPort(), () -> randomIntBetween(49152, 65535));
```

Comment:
I wonder if this will randomly select another open port one day and cause a flap. Not sure what the best approach is here.

Reply:
Addressed in a3d37d55bf16be0be79c2e548283d8c1c6f96ba1